datasets/smartdata/smartdata.py

```diff
@@ -0,0 +1,210 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""\
+DFKI SmartData Corpus is a dataset of 2598 German-language documents
+which has been annotated with fine-grained geo-entities, such as streets,
+stops and routes, as well as standard named entity types."""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+from json import JSONDecodeError, JSONDecoder
+
+import datasets
+
+
+_CITATION = """\
+@InProceedings{SCHIERSCH18.85,
+  author = {Martin Schiersch and Veselina Mironova and Maximilian Schmitt and Philippe Thomas and Aleksandra Gabryszak and Leonhard Hennig},
+  title = "{A German Corpus for Fine-Grained Named Entity Recognition and Relation Extraction of Traffic and Industry Events}",
+  booktitle = {Proceedings of the Eleventh International Conference on Language Resources and Evaluation (LREC 2018)},
+  year = {2018},
+  month = {May 7-12, 2018},
+  address = {Miyazaki, Japan},
+  editor = {Nicoletta Calzolari (Conference chair) and Khalid Choukri and Christopher Cieri and Thierry Declerck and Sara Goggi and Koiti Hasida and Hitoshi Isahara and Bente Maegaard and Joseph Mariani and Hélène Mazo and Asuncion Moreno and Jan Odijk and Stelios Piperidis and Takenobu Tokunaga},
+  publisher = {European Language Resources Association (ELRA)},
+  isbn = {979-10-95546-00-9},
+  language = {english}
+  }
+"""
+
+_DESCRIPTION = """\
+DFKI SmartData Corpus is a dataset of 2598 German-language documents
+which has been annotated with fine-grained geo-entities, such as streets,
+stops and routes, as well as standard named entity types. It has also
+been annotated with a set of 15 traffic- and industry-related n-ary
+relations and events, such as Accidents, Traffic jams, Acquisitions,
+and Strikes. The corpus consists of newswire texts, Twitter messages,
+and traffic reports from radio stations, police and railway companies.
+It allows for training and evaluating both named entity recognition
+algorithms that aim for fine-grained typing of geo-entities, as well
+as n-ary relation extraction systems."""
+
+_HOMEPAGE = "https://www.dfki.de/web/forschung/projekte-publikationen/publikationen-uebersicht/publikation/9427/"
+
+_LICENSE = "CC-BY 4.0"
+
+_URLs = {
+    "train": "https://github.com/DFKI-NLP/smartdata-corpus/raw/master/v3_20200302/train.json.gz",
+    "dev": "https://github.com/DFKI-NLP/smartdata-corpus/raw/master/v3_20200302/dev.json.gz",
+    "test": "https://github.com/DFKI-NLP/smartdata-corpus/raw/master/v3_20200302/test.json.gz",
+}
+
+
+class SmartData(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class Smartdata(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `smartdata`